Documented the search indexer abstraction introduced in Contao 4.9 (#212)
contao/docs: docs/dev/framework/search-indexing.md

  
Indexing using the `SearchIndexListener` has the advantage that it happens behind the scenes without you having to worry
about it. It doesn't require any cron job and it makes sure the search index is always updated so that when your editors
edit a page, it's indexed as soon as the first visitor visits the modified page. Moreover, if a request is not successful
(meaning the HTTP status code is not in the `2xx` range), it will also delete that URI from the search index.
However, it also comes with the disadvantage that it happens on every single page request which may affect the
performance of your website.
Although the `kernel.terminate` event is triggered **after** the response has been sent to the browser, this only works
if you are running PHP using `php-fpm`. In case of `mod_php` or `fcgi` this will happen **before** the response is sent
to the browser, which may extend the time needed to finish the response.

This is why you can configure the behaviour of the `SearchIndexListener` as follows:

```yaml
# config/config.yml
contao:
    search:
        listener:
            index: true # Configure whether you want to update the index entry on every request

>             delete: false # Configure whether you want to delete the obsolete entry on every request

Reviewer: ```suggestion
            delete: false # Configure whether you want to delete an entry if a request is not successful
```
